includes/page-assets-config.php

```diff
@@ -0,0 +1,298 @@
+<?php
+/**
+ * Class that handles config options
+ **/
+if ( ! class_exists( 'UCF_Page_Assets_Config' ) ) {
+    class UCF_Page_Assets_Config {
+        public static
+            $option_prefix   = 'ucf_page_assets_',
+            $option_defaults = array(
+                'enabled_post_types' => array(
+                    'page' => 'on'
+                )
+            );
+
+        /**
+		 * Creates options via the WP Options API that are utilized by the
+		 * plugin. Intended to be run on plugin activation.
+		 *
+		 * @author Jim Barnes, Jo Dickson
+		 * @since 1.0.0
+		 **/
+        public static function add_options() {
+            $defaults = self::$option_defaults;
+
+            add_option( self::$option_prefix . 'enabled_post_types', $defaults['enabled_post_types'] );
+        }
+
+        /**
+		 * Deletes options via the WP Options API that are utilized by the
+		 * plugin. Intented to be run on plugin deactivation.
+		 * @author Jim Barnes
+		 * @since 1.0.0
+		 **/
+		public static function delete_options() {
+			delete_option( self::$option_prefix . 'enabled_post_types' );
+		}
+
+        /**
+		 * Returns a list of default plugin options. Applies any overridden
+		 * default values set within the options page.
+		 *
+		 * @author Jim Barnes, Jo Dickson
+		 * @since 1.0.0
+		 *
+		 * @return array
+		 **/
+		public static function get_option_defaults() {
+			$defaults = self::$option_defaults;
+			$configurable_defaults = array(
+				'enabled_post_types' => get_option( self::$option_prefix . 'enabled_post_types' )
+			);
+			$configurable_defaults = self::format_options( $configurable_defaults );
+			$default = array_merge( $defaults, $configurable_defaults );
+			return $defaults;
+		}
+
+		/**
+		 * Returns an array with plugin defaults applied.
+		 *
+		 * @author Jo Dickson
+		 * @since 1.0.0
+		 *
+		 * @param array $list
+		 * @param boolean $list_keys_only Modifies results to only return array key
+		 *                                values present in $list.
+		 * @return array
+		 **/
+		public static function apply_option_defaults( $list, $list_keys_only=false ) {
+			$defaults = self::get_option_defaults();
+			$options = array();
+			if ( $list_keys_only ) {
+				foreach( $list as $key => $val ) {
+					$options[$key] = !empty( $val ) ? $val : $defaults[$key];
+				}
+			} else {
+				$options = array_merge( $defaults, $list );
+			}
+			$options = self::format_options( $options );
+			return $options;
+		}
+
+        /**
+		 * Performs typecasting, sanitization, etc on an array of plugin options.
+		 *
+		 * @author Jim Barnes, Jo Dickson
+		 * @since 1.0.0
+		 *
+		 * @param array $list
+		 * @return array
+		 **/
+		public static function format_options( $list ) {
+			foreach( $list as $key => $val ) {
```

Comment:
Consider taking out this foreach loop (or comment it) if it's not currently being used